webapp/channels/src/components/user_settings_modal/advanced_display_settings/utils.tsx

```diff
@@ -0,0 +1,307 @@
+                    id='user.settings.advance.on'
+                    defaultMessage='On'
+                />
+            ),
+            name: `${AdvanceSettings.JOIN_LEAVE}`,
+            key: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.ON}`,
+            value: OnAndOff.ON,
+        },
+        {
+            dataTestId: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.OFF}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.off'
+                    defaultMessage='Off'
+                />
+            ),
+            name: `${AdvanceSettings.JOIN_LEAVE}`,
+            key: `${AdvanceSettings.JOIN_LEAVE}-${OnAndOff.OFF}`,
+            value: OnAndOff.OFF,
+        },
+    ],
+};
+
+export const SyncedDraftInputFieldData: FieldsetRadio = {
+    options: [
+        {
+            dataTestId: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.ON}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.on'
+                    defaultMessage='On'
+                />
+            ),
+            name: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}`,
+            key: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.ON}`,
+            value: OnAndOff.ON,
+        },
+        {
+            dataTestId: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.OFF}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.off'
+                    defaultMessage='Off'
+                />
+            ),
+            name: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}`,
+            key: `${AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED}-${OnAndOff.OFF}`,
+            value: OnAndOff.OFF,
+        },
+    ],
+};
+
+export const UnreadScrollPositionInputFieldData: FieldsetRadio = {
+    options: [
+        {
+            dataTestId: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.ON}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.startFromLeftOff'
+                    defaultMessage='Start me where I left off'
+                />
+            ),
+            name: `${AdvanceSettings.UNREAD_SCROLL_POSITION}`,
+            key: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.ON}`,
+            value: UnreadScrollPosition.LEFT,
+        },
+        {
+            dataTestId: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.OFF}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.startFromNewest'
+                    defaultMessage='Start me at the newest message'
+                />
+            ),
+            name: `${AdvanceSettings.UNREAD_SCROLL_POSITION}`,
+            key: `${AdvanceSettings.UNREAD_SCROLL_POSITION}-${OnAndOff.OFF}`,
+            value: UnreadScrollPosition.NEWEST,
+        },
+    ],
+};
+
+export const getCtrlSendText = () => {
```

Comment:
I don't think we need to support being able to change from Mac to Windows, so I think we could probably make this a constant instead of making it a function. That way, we aren't returning a new object in case we were to pass these message descriptors as props into something else